### PR TITLE
Deprecated doctrine classes

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Deprecated
 - Deprecated `Model/Adapter/AdapterChain` 
 - Deprecated `Model/Adapter/AdapterInterface`
 - Deprecated `Model/Adapter/DoctrineORMAdapter`
 - Deprecated `Model/Adapter/DoctrinePHPCRAdapter`
 - Deprecated `Model/BaseDocumentManager`
 - Deprecated `Model/BaseEntityManager`
 - Deprecated `Model/BaseManager`
 - Deprecated `Model/BasePHPCRManager`
 - Deprecated `Model/ManagerInterface`
 - Deprecated `Model/PageableManagerInterface`
```

## Subject

Finally, we can deprecate (and remove) some classes.

### CODE REVIEW

**Comment by core23** [comment]
Can someone help with the deprecations? @greg0ire?

**Comment by core23** [comment]
Or maybe @kunicmarko20 ?

**Comment by kunicmarko20** [comment]
help with what?

**Comment by core23** [comment]
Tests are failing because of the deprecations :/

**Comment by greg0ire** [comment]
I might be able to help this week-end

**Comment by greg0ire** [comment]
Using a regex, I can get a stack trace:
```
SYMFONY_DEPRECATIONS_HELPER='/BaseDocument/' ./phpunit

Remaining deprecation triggered by Composer\Autoload\ClassLoader::loadClass:
The Sonata\CoreBundle\Model\BaseDocumentManager class is deprecated since version 3.x and will be removed in 4.0.
Stack trace:
#0 src/Model/BaseDocumentManager.php(16): trigger_error('The Sonata\\Core...', 16384)
#1 vendor/composer/ClassLoader.php(444): include('/home/greg/dev/...')
#2 vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('/home/greg/dev/...')
#3 [internal function]: Composer\Autoload\ClassLoader->loadClass('Sonata\\CoreBund...')
#4 {main}
```

Not very helpful I'm afraid… not sure yet what triggers this autoloading.

**Comment by greg0ire** [comment]
Ah I think I know… the class that extends it for test purposes is outside the test.

**Comment by greg0ire** [comment]
I will simply move them outside, since I cannot use anonymous classes yet

**Comment by greg0ire** [comment]
It works, expect a PR soon

**Comment by greg0ire** [comment]
and/or allow me to push here somehow

**Comment by core23** [comment]
Thanks @greg0ire ❤️

File: src/Model/Adapter/AdapterChain.php [hunk 1]
```diff
@@ -11,34 +11,14 @@
...
-        foreach ($this->adapters as $adapter) {
-            $identifier = $adapter->getNormalizedIdentifier($model);
-
-            if ($identifier) {
-                return $identifier;
-            }
-        }
-    }
+@trigger_error(
+    'The '.__NAMESPACE__.'\AdapterInterface class is deprecated since version 3.x and will be removed in 4.0.',
```
[the remark this answers is not in the document]
Comment (core23) [reply src/Model/Adapter/AdapterChain.php]: Done

**Pull request merged**